sisl/viz/plotly/plots/fatbands.py

```diff
@@ -215,10 +215,10 @@ def _read_siesta_output(self, wfsx_file, bands_file, root_fdf):
 
         # If the wfsx doesn't exist, we will not even bother to read the bands
         if not wfsx_file.exists():
-            raise Exception(f"We didn't find a WFSX file in the location {wfsx_file}")
+            raise ValueError(f"We did not find a WFSX file in the location {wfsx_file}")
 
         # Otherwise we will make BandsPlot read the bands
-        BandsPlot._read_siesta_output(self)
+        super()._read_siesta_output()
```

Comment:
Yes and no. I see your point. However, this makes it clear that it is something in the inheritance diagram that gets called. I.e. not some 3rd party class.

Reply:
Ok, but it may be harder to debug if `_read_siesta_output` is renamed or something like that. Also a new developer will have to check the inheritance tree of the class to see where that function appears first. But yeah what you say is also true.

Although I don't understand why you are so against using methods that are not in the class inheritance tree. I'm not saying you should do it all the time, but I don't understand why is it that bad that you have to use `super()` here so that nobody thinks that you are doing it.

E.g. isn't it fine that an external function modifies the plot object?
```
def act(plot):
    plot.attribute = 2
``` 

I agree that it is better to not mix one style of coding with the other, but I don't see it so bad that I can not write `BandsPlot._read_siesta_output(self)` just in case.

Reply:
I am against using class methods not in the inheritance tree. Regular methods are fine to use. Although you suggestion there I don't particularly like since it alters the object in-place. For setting up stuff this may be ok. But for *while* running I am not too fond. But then again, if it helps and aids understanding then ok. But one shouldn't save 2 lines of code that obfuscates reading and program flow. :)